generated/nidaqmx/_base_interpreter.py

```diff
@@ -0,0 +1,1885 @@
+
+    @abc.abstractmethod
+    def get_timing_attribute_int32(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_timing_attribute_string(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_timing_attribute_uint32(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_timing_attribute_uint64(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_trig_attribute_bool(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_trig_attribute_double(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_trig_attribute_double_array(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_trig_attribute_int32(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_trig_attribute_int32_array(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_trig_attribute_string(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_trig_attribute_uint32(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_watchdog_attribute_bool(self, task, lines, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_watchdog_attribute_double(self, task, lines, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_watchdog_attribute_int32(self, task, lines, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_watchdog_attribute_string(self, task, lines, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_write_attribute_bool(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_write_attribute_double(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_write_attribute_int32(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_write_attribute_string(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_write_attribute_uint32(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_write_attribute_uint64(self, task, attribute):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def is_task_done(self, task):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def load_task(self, session_name):
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def read_analog_f64(
+            self, task, num_samps_per_chan, timeout, fill_mode,
+            array_size_in_samps):
+        raise NotImplementedError
```

Comment:
Compare to `_read_analog_f_64`: https://github.com/ni/nidaqmx-python/blob/master/generated/nidaqmx/_task_modules/read_functions.py

`_read_analog_f_64` takes a numpy array parameter called `read_array`:
- This allows it to use a preallocated array.
- The numpy array knows its size, so `_read_analog_f_64` doesn't take `array_size_in_samps`.

Reply:
Here is where the preallocated array is used: `task.in_stream.readinto()`: https://github.com/ni/nidaqmx-python/blob/master/generated/nidaqmx/_task_modules/in_stream.py#L2371

Reply:
Ok. I have added the array parameters for functions that has mechanism - `passed-in`.
Also, removed the size parameter `array_size_in_samps` in `read_analog_f64` using **mechanism** of the array parameter.